Automattic/jetpack
# Add recommendation for anti-spam

#### Changes proposed in this Pull Request:

* This PR adds a new recommendation that provides some information about managing spam comments that shows up when a post receives 5+ comments.

#### Jetpack product discussion


pbNhbs-1iw-p2

#### Does this pull request change what data or activity we track or use?


This PR adds a new resource recommendation step and uses existing analytics events introduced in #23600 to record interactions with the recommendation on the individual step screen and on the summary screen. Analytics events that can be recorded for this new step are:
```
// Individual Step:
jetpack_recommended_resource_learn_more_click
jetpack_recommended_resource_read_click
jetpack_recommended_resource_skip_click

// Summary Screen:
jetpack_recommendations_summary_learn_more_click
```

#### Testing instructions:






To reset the assistant state while testing, you can use the following commands ( you can run these with `jetpack docker` if using the Docker development environment )
* `wp jetpack options delete recommendations_data`
* `wp jetpack options delete recommendations_conditional`
* `wp jetpack options delete recommendations_step`

Note that if your test site has the Akismet plugin enabled, you will not receive the new recommendation as outlined in the test plan below.

---

* Checkout this branch on your Jetpack test site.
* Visit the Jetpack dashboard. Note that the "Recommendations" tab in the dashboard navigation does not show any badge indicating that there is a new recommendation.
![Screen Shot 2022-03-21 at 2 27 58 PM](https://user-images.githubusercontent.com/18016357/159340092-2adb1a7e-7834-4986-b965-33519fa1ee11.png)

* Now, visit the front end of the site and go to an individual post.
* Add sample comments to the post until the post has 5 or more comments.
* Navigate back to the Jetpack Dashboard. You should see a red badge with a "1" next to the Recommendations tab in the navigation on the Jetpack dashboard - indicating a new recommendation is available.
![Screen Shot 2022-03-11 at 4 00 23 PM](https://user-images.githubusercontent.com/18016357/158703217-3b7bb9bb-a1e9-4b6a-814e-d81e167094f1.png)

* Click on the "Recommendations" tab in the navigation - you should now see a recommendation step that provides some information about spam comments. There should be a green "New" badge that shows on this step in the upper-left corner. The CTA button should link out to this article: https://jetpack.com/2021/04/29/why-spam-comments-exist-and-how-to-stop-them/
![Screen Shot 2022-03-30 at 9 49 08 AM](https://user-images.githubusercontent.com/18016357/160915195-6e35456f-5f3f-49f7-bac0-176b03bf018d.png)

* Click on "Read Later"  to proceed to the summary screen. On the summary screen, you should see the a new "Resources" section with the "Spam Management" recommendation listed. A green "New" badge should show next to "Spam Management" in the list. The "Read More" button should link to the same article mentioned in the previous step.
![Screen Shot 2022-03-30 at 9 48 54 AM](https://user-images.githubusercontent.com/18016357/160915392-5b10c86b-4ba4-4810-aab9-9e1fd02c4361.png)

* If you reload the page on the summary screen, the green "New" badge next to Site Security in the summary resources list and the red "1" indicator next to "Recommendations" in the dashboard navigation  should no longer show.
* Automated unit tests should be passing
* Visuals should match this reference: G151IYl8Lm4f0NjgLpR943-fi-2132%3A39570
